src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -38,7 +36,7 @@
   private final long mSequenceNumber;
   private final Operation[] mOperations;
   private final Memo mMemo;
-  private final TimeBounds mTimeBounds;
+  private final PreconditionsV2 preconditions;
```

Comment:
We shouldn't use the XDR type here, but rather a user-friendly wrapper like we do in the Go SDK (same way we use `long` over `xdr.SequenceNumber` etc. here, see tangential convo on Discord)

Reply:
Plus, a transaction can contain V1 (timebounds only), V2 (timebounds + others), or even no preconditions, so this wouldn't capture the other cases.

Reply:
yeah, I [wondered about that ](https://github.com/stellar/go/pull/4303/files#r834845705), I'll refactor conditions into non-xdr data transfer object. 

The V1/V2/None conditions emitted in tx xdr will be fleshed out just a little more, I've got a [todo there now](https://github.com/stellar/java-stellar-sdk/pull/415/files#diff-54693c2dd591263011d0ad426cbf1f937b377134d13ce83bcc96977d38270b6cR214)

Reply:
created wrapper [TransactionPreconditions](https://github.com/stellar/java-stellar-sdk/pull/415/files#diff-5ae3e337d7c0699e36fe2fec0ebe63510d548b2057bd400dc80186b8104042a3) to proxy the xdr.Preconditions.